### ISSUE TITLE
On A64FX can't setup counter

### ISSUE THREAD

**Issue by vchuravy** [JuliaPerf/LIKWID.jl#45]
cc: @giordano

On Ookami running `likwid-perfctr` works:

```
[vchuravy@fj-debug1 ~]$ likwid-perfctr -m  -C 0 -g FLOPS_DP julia perfctr.jl
--------------------------------------------------------------------------------
CPU name:	
CPU type:	Fujitsu A64FX
CPU clock:	0.00 GHz
--------------------------------------------------------------------------------
--------------------------------------------------------------------------------
Region saxpy, Group 1: FLOPS_DP
+-------------------+------------+
|    Region Info    | HWThread 0 |
+-------------------+------------+
| RDTSC Runtime [s] |   0.100502 |
|     call count    |          1 |
+-------------------+------------+

+----------------------+---------+------------+
|         Event        | Counter | HWThread 0 |
+----------------------+---------+------------+
|     INST_RETIRED     |   PMC0  |   95528990 |
|      CPU_CYCLES      |   PMC1  |  179848400 |
| FP_DP_FIXED_OPS_SPEC |   PMC3  |      20114 |
| FP_DP_SCALE_OPS_SPEC |   PMC4  |          0 |
+----------------------+---------+------------+

+--------------------------+------------+
|          Metric          | HWThread 0 |
+--------------------------+------------+
|    Runtime (RDTSC) [s]   |     0.1005 |
|        Clock [MHz]       |  1789.5007 |
|            CPI           |     1.8827 |
|     DP (FP) [MFLOP/s]    |     0.2001 |
| DP (FP+SVE128) [MFLOP/s] |     0.2001 |
| DP (FP+SVE256) [MFLOP/s] |     0.2001 |
| DP (FP+SVE512) [MFLOP/s] |     0.2001 |
+--------------------------+------------+

```
But using the CLI interface doesn't

```
julia> using LIKWID

julia> LIKWID.LibLikwid.likwid_pinThread(0)
1

julia> PerfMon.init(0)
true

julia> groupid = PerfMon.add_event_set("FLOPS_DP")
1

julia> PerfMon.setup_counters(groupid)
Setup of counters failed for thread 0
false
```

**Comment by TomTheBear** [comment]
Can you please re- run the setup_counters call with verbosity level 3.

**Comment by giordano** [comment]
```julia
julia> LIKWID.LibLikwid.perfmon_setVerbosity(3)

julia> PerfMon.setup_counters(groupid)
DEBUG - [perfmon_setupCountersThread_perfevent:662] PID of application required. Use LIKWID_PERF_PID env variable or likwid-perfctr options
Setup of counters failed for thread 0
false
```

**Comment by giordano** [comment]
Judging by the debug message I presume we should do something like
```julia
julia> ENV["LIKWID_PERF_PID"] = getpid()
2762058

julia> PerfMon.setup_counters(groupid)
DEBUG - [perfmon_setupCountersThread_perfevent:665] Using PID 2762058 for perf_event measurements
DEBUG - [perfmon_setupCountersThread_perfevent:741] SETUP_PMC [0] Register 0x0 , Flags: 0x8 
DEBUG - [perfmon_setupCountersThread_perfevent:872] perf_event_open: cpu_id=0 pid=2762058 flags=0
DEBUG - [perfmon_setupCountersThread_perfevent:741] SETUP_PMC [0] Register 0x1 , Flags: 0x11 
DEBUG - [perfmon_setupCountersThread_perfevent:872] perf_event_open: cpu_id=0 pid=2762058 flags=0
DEBUG - [perfmon_setupCountersThread_perfevent:741] SETUP_PMC [0] Register 0x3 , Flags: 0x80C7 
DEBUG - [perfmon_setupCountersThread_perfevent:872] perf_event_open: cpu_id=0 pid=2762058 flags=0
DEBUG - [perfmon_setupCountersThread_perfevent:741] SETUP_PMC [0] Register 0x4 , Flags: 0x80C6 
DEBUG - [perfmon_setupCountersThread_perfevent:872] perf_event_open: cpu_id=0 pid=2762058 flags=0
true
```
?

**Comment by vchuravy** [comment]
Ok so the issue is that `perf_event_paranoid > 0` thus we can't use allpid. 

https://github.com/RRZE-HPC/likwid/blob/d82c671fcbb60915bf148769e5934f2cff713a35/src/libperfctr.c

```
julia> using LIKWID

julia> LIKWID.accessmode()
ACCESSMODE_PERF::AccessMode = -1
```

**Comment by carstenbauer** [comment]
FYI: I triggered a [new release](https://github.com/JuliaRegistries/General/pull/73481) (0.4.3) with the proposed fix implemented in #47.

**Comment by vchuravy** [comment]
Thanks!
